freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -70,11 +68,37 @@ def handle(self, event):
             self.log_info(msg)
             return []
 
-        # Get original nvrs of all builds in the advisory
-        original_nvrs = [build.original_nvr for build in db_event.builds]
+        # Get builds NVRs from the advisory attached to the message/event
+        affected_nvrs = set()
+        for product_info in event.advisory.builds.values():
+            for build in product_info['builds']:
+                affected_nvrs.add(build['nvr'])
+
+        # Search for the first event that triggered the chain of rebuilds
+        # to get original NVRs from that build event
+        old_db_events_list = []
+        for original_nvr in affected_nvrs:
+            olb_db_event = Event.get_original_build_by_nvr(
+                db.session, original_nvr)
+
+            if olb_db_event is None:
+                msg = f"Can't find old build event for nvr : {original_nvr}"
+                self.log_warn(msg)
+                continue
+            old_db_events_list.append(olb_db_event)
+        if not old_db_events_list:
+            msg = f"Can't find any previous build event for nvrs: {affected_nvrs}"
+            self.log_warn(msg)
+            return []
+
+        # Get original nvrs of all old builds of rpms in the advisory
```

Comment:
An advisory filed by BOTAS can contains only part of the rebuilt images in an event, we only want the original NVRs for the shipped NVRs in this advisory, not all the original NVRs in the event.

Reply:
Yes you are right, didn't think about it. I will fix it.